include/operator.h

```diff
@@ -555,3 +555,594 @@ class Operator : public Subscriptor
+              if constexpr (n_components == 1)
+                {
+                  VectorizedArray<number> coeff = 0;
+
+                  const auto point_batch = phi.quadrature_point(q);
+
+                  for (unsigned int v = 0; v < VectorizedArray<number>::size();
+                       ++v)
+                    {
+                      Point<dim> single_point;
+                      for (unsigned int d = 0; d < dim; d++)
+                        single_point[d] = point_batch[d][v];
+                      coeff[v] = rhs_func->value(single_point);
+                    }
+
+                  phi.submit_value(coeff, q);
+                }
+              else
+                {
+                  Tensor<1, n_components, VectorizedArray<number>>
+                    temp; // n_components == dim
+                  for (unsigned int v = 0; v < VectorizedArray<number>::size();
+                       ++v)
+                    {
+                      for (int i = 0; i < n_components; ++i)
+                        temp[i][v] = 1.0;
+                    }
+
+                  phi.submit_value(temp, q);
+                }
+
+            phi.integrate_scatter(EvaluationFlags::values, dst);
+          }
+      },
+      system_rhs,
+      dummy,
+      true);
+
+    AffineConstraints<number> constraints_without_dbc;
+    constraints_without_dbc.reinit(
+      DoFTools::extract_locally_relevant_dofs(dof_handler));
+    DoFTools::make_hanging_node_constraints(dof_handler,
+                                            constraints_without_dbc);
+    constraints_without_dbc.close();
+
+    VectorType b, x;
+
+    this->initialize_dof_vector(b);
+    this->initialize_dof_vector(x);
+
+    typename MatrixFree<dim, number>::AdditionalData data;
+    data.mapping_update_flags =
+      update_values | update_gradients | update_quadrature_points;
+
+    MatrixFree<dim, number> matrix_free;
+    matrix_free.reinit(
+      mapping, dof_handler, constraints_without_dbc, quad, data);
+
+    // set constrained
+    constraints.distribute(x);
+
+    // perform matrix-vector multiplication (with unconstrained system and
+    // constrained set in vector)
+    matrix_free.cell_loop(
+      &ElasticityOperator::do_cell_integral_range, this, b, x, true);
+
+    // clear constrained values
+    constraints.set_zero(b);
+
+    // move to the right-hand side
+    system_rhs -= b;
+  }
+
+private:
+  void
+  do_cell_integral_local(FECellIntegrator &integrator) const
+  {
+    integrator.evaluate(EvaluationFlags::gradients);
+
+    for (unsigned int q = 0; q < integrator.n_q_points; ++q)
+      {
+        SymmetricTensor<2, dim, VectorizedArray<number>> sym_grad_u =
+          2. * mu * integrator.get_symmetric_gradient(q);
+        const auto lambda_div_u = lambda * integrator.get_divergence(q);
+        // Add lambda div(u)I
+        for (unsigned int i = 0; i < dim; ++i)
+          sym_grad_u[i][i] += lambda_div_u;
```

Comment:
I would do this slightly differently to avoid querying the evaluator for the second time:
```suggestion
        const SymmetricTensor<2, dim, VectorizedArray<number>> sym_grad_u =
          integrator.get_symmetric_gradient(q);
        const auto lambda_div_u = lambda * trace(sym_grad_u);
        const auto stress = 2. * mu * sym_grad_u;
        // Add lambda div(u)I
        for (unsigned int i = 0; i < dim; ++i)
          stress[i][i] += lambda_div_u;
```
This creates another temporary, but a good compiler should do that all in registers.